src/lib/Microsoft.Health.Fhir.R4.Ingest/Service/R4DeviceAndPatientLookupIdentityService.cs

```diff
@@ -48,5 +49,20 @@ protected async override Task<(string DeviceId, string PatientId)> LookUpDeviceA
             var device = await ResourceManagementService.GetResourceByIdentityAsync<Model.Device>(value, system).ConfigureAwait(false) ?? throw new FhirResourceNotFoundException(ResourceType.Device);
             return (device.Id, GetPatientIdFromDevice(device));
         }
+
+        protected async override Task<IDictionary<ResourceType, string>> ResolveResourceIdentitiesInternalAsync(IMeasurementGroup input)
+        {
+            EnsureArg.IsNotNull(input, nameof(input));
+
+            var identities = await base.ResolveResourceIdentitiesInternalAsync(input).ConfigureAwait(false);

```

Comment:
`LookUpDeviceAndPatientIdAsync` called by `base.ResolveResourceIdentitiesInternalAsync` can throw a `FhirResourceNotFoundException`. Should we handle this exception?

Reply:
That is intentional, if the patient, device, or the encounter id fails to resolve on the FHIR service a FhirResourceNotFoundException is expected.